src/stan/mcmc/base_mcmc.hpp

```diff
@@ -42,14 +46,8 @@ namespace stan {
 
     protected:
       std::string name_;
```

Comment:
We should define name_ on construction and have name get passed through all of the MCMC derivations to satisfy the "make all of the things defined on construction" idiom, not how I originally did it.  If you want to avoid this here then I'd make it an issue.